internal/graph/builder.go

```diff
@@ -0,0 +1,101 @@
+package graph
+
+type CheckResolverBuilder struct {
+	localCheckerOpts       []LocalCheckerOption
+	cacheOpts              []CachedCheckResolverOpt
+	dispatchThrottlingOpts []DispatchThrottlingCheckResolverOpt
+
+	cycleDetectionCheckResolver     *CycleDetectionCheckResolver
+	cachedCheckResolver             *CachedCheckResolver
+	dispatchThrottlingCheckResolver *DispatchThrottlingCheckResolver
+	localCheckResolver              *LocalChecker
+}
```

Comment:
In your example you still have this right

```
	first := &CachedResolver{9, nil}
	second := &DispatchThrottlerResolver{10000, nil}
	first.SetDelegate(second)
	second.SetDelegate(first)
```

This wouldn't be a runtime change right? We do the same construction right now too.  

I understand this could be easier if we do `SetDelegate` as well using the CircularList then construction might be easier